python/ray/data/impl/pandas_block.py

```diff
@@ -164,6 +174,55 @@ def _empty_table() -> "pandas.DataFrame":
     def _sample(self, n_samples: int, key: "SortKeyT") -> "pandas.DataFrame":
         return self._table[[k[0] for k in key]].sample(n_samples, ignore_index=True)
 
+    def count(self, on: KeyFn, ignore_nulls: bool) -> Optional[U]:
+        col = self._table[on]
```

Comment:
Again if 'on' is a function, this seems to break.

Reply:
We're guaranteed that `on` keys are strings for tabular blocks by upstream validation, but we can add an extra check here.